seed/api/v2_1/views.py

```diff
@@ -117,8 +123,131 @@ def get_queryset(self):
+        if os.path.exists(bs_file):
+            bs.import_file(bs_file)
+            xml = bs.export(property_view.state, BuildingSync.BRICR_STRUCT)
+            return HttpResponse(xml, content_type='application/xml')
+        else:
+            return JsonResponse({
+                'success': False,
+                'message': "Could not find valid BuildingSync file attached to PropertyState"
+            })
 
+    @detail_route(methods=['PUT'])
+    def update_with_building_sync(self, request, pk):
         """
-        return JsonResponse(
-            {"status": "error", "message": "Not yet implemented. PK was {}".format(pk)})
+        Does not work in Swagger!
+
+        Update an existing PropertyView with a building file. Currently only supports BuildingSync.
+        ---
+        consumes:
+            - multipart/form-data
+        parameters:
+            - name: pk
+              description: The PropertyView to update with this buildingsync file
+              type: path
+              required: true
+            - name: organization_id
+              type: integer
+              required: true
+            - name: cycle_id
+              type: integer
+              required: true
+            - name: file_type
+              type: string
+              enum: ["Unknown", "BuildingSync", "GeoJSON"]
+              required: true
+            - name: file
+              description: In-memory file object
+              required: true
+              type: file
+        """
+        if len(request.FILES) == 0:
+            return JsonResponse({
+                'success': False,
+                'message': "Must pass file in as a Multipart/Form post"
+            })
+
+        the_file = request.data['file']
+        file_type = BuildingFile.str_to_file_type(request.data.get('file_type', 'Unknown'))
+        organization_id = request.data['organization_id']
+        cycle_id = request.data.get('cycle_id', None)
+
+        if not cycle_id:
+            return JsonResponse({
+                'success': False,
+                'message': "Cycle ID is not defined"
+            })
+        else:
+            cycle = Cycle.objects.get(pk=cycle_id)
+
+        building_file = BuildingFile.objects.create(
+            file=the_file,
+            filename=the_file.name,
+            file_type=file_type,
+        )
+
+        try:
+            property_view = PropertyView.objects.get(pk=pk, cycle=cycle)
+        except PropertyView.DoesNotExist:
+            return JsonResponse({
+                'success': False,
+                'message': 'Cannot match a PropertyView with pk=%s; cycle_id=%s' % (pk, cycle_id)
+            })
+
+        # passing in the property view pk should allow it to process the buildingsync file but not create a new PV
+        p_status, property_view, messages = building_file.process(organization_id, cycle,
+                                                                  property_view=property_view)
+
```

Comment:
So at this point, we have a property_view that we got from the process() method.  But if I'm understanding correctly, that doesn't actually point to a valid property_state behind the scenes.  It's in a weird state.  Let me know if my path forward sounds correct:
 - I should still look up the current property_view.
 - Then get the current property_state from the view
 - Then do a deepcopy of the property_state
 - Call the process method, but pass in a property_state instance (instead of a property_view), and have it process the file and apply it to the passed in property_state
 - Once back in this method, do some data persistence:
   - PropertyState.extra_data should be persisted because of the deepcopy I think
   - BuildingFile.property_state - is this BuildingFile instance the original file that imported the building, or the new BuildingSync file being added?  Still a little fuzzy here
   - Scenario.property_state - keep the scenario ID from the original property view/state, and update that scenario with the newly created property state
   - Simulation.property_state - same
 - Then promote the newly created property state to a property view
 - Done?

Reply:
talk this morning?